pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -98,14 +98,20 @@ func validNodeLabels(proposed esv1.Elasticsearch, exposedNodeLabels NodeLabels)
 	return errs
 }
 
-func check(es esv1.Elasticsearch, validations []validation) field.ErrorList {
+func check(es esv1.Elasticsearch, validations []validation) (string, field.ErrorList) {
 	var errs field.ErrorList
 	for _, val := range validations {
 		if err := val(es); err != nil {
 			errs = append(errs, err...)
 		}
 	}
-	return errs
+
+	// check if Elasticsearch version is deprecated
+	warnings, deprecatedErrors := checkDeprecatedVersion(&es)
```

Comment:
I think we can remove `checkDeprecatedVersion(...)`?
```suggestion
warnings, deprecatedErrors := commonv1.CheckDeprecatedStackVersion(es.Spec.Version, version.DeprecatedVersions)
```